src/resources/2022/Stillsuit.ts

```diff
@@ -56,3 +57,93 @@ export function distillateModifier(modifier: NumericModifier): number {
   // If a match is found, parse and return the captured number; otherwise, return 0
   return match ? Number(match[1]) : 0;
 }
+
+type StillsuitTag = Exclude<FamiliarTag, "pokefam">;
+
+export const MODIFIER_TAGS: Record<StillsuitTag, NumericModifier> =
+  Object.freeze({
+    mineral: "Muscle",
+    robot: "Muscle",
+    organic: "Muscle",
+    hasbones: "Muscle",
+    technological: "Mysticality",
+    orb: "Mysticality",
+    sentient: "Mysticality",
+    polygonal: "Mysticality",
+    software: "Mysticality",
+    cantalk: "Mysticality",
+    humanoid: "Moxie",
+    hashands: "Moxie",
+    cute: "Moxie",
+    good: "Moxie",
+    phallic: "Moxie",
+    animatedart: "Moxie",
+    person: "Moxie",
+    haseyes: "Item Drop",
+    object: "Item Drop",
+    haslegs: "Item Drop",
+    food: "Food Drop",
+    vegetable: "Food Drop",
+    edible: "Food Drop",
+    animal: "Damage Reduction",
+    insect: "Damage Reduction",
+    wearsclothes: "Damage Reduction",
+    isclothes: "Damage Reduction",
+    hasshell: "Damage Reduction",
+    haswings: "Initiative",
+    fast: "Initiative",
+    flies: "Initiative",
+    hovers: "Initiative",
+    swims: "Initiative",
+    aquatic: "Initiative",
+    spooky: "Spooky Damage",
+    undead: "Spooky Damage",
+    evil: "Spooky Damage",
+    reallyevil: "Spooky Damage",
+    hot: "Hot Damage",
+    cold: "Cold Damage",
+    sleaze: "Sleaze Damage",
+    stench: "Stench Damage",
+    bite: "Weapon Damage",
+    hasclaws: "Weapon Damage",
+    hasbeak: "Weapon Damage",
+    hasstinger: "Weapon Damage",
+    hard: "Weapon Damage",
+  } as const);
+
+function isStillsuitTag(tag: string): tag is StillsuitTag {
+  return tag in MODIFIER_TAGS;
+}
+
+/**
+ * Calculate the ratio of stillsuit modifiers for a particular familiar.
+ * @param familiar The familiar in question
+ * @returns An object whose keys are NumericModifiers potentially granted by the stillsuit distillate from this familiar, and whose values are the relative weights of those modifiers
+ */
+export function modifierRatio(
+  familiar: Familiar,
+): Partial<Record<NumericModifier, number>> {
+  const tags = getFamiliarTags(familiar);
+  return tags
+    .filter(isStillsuitTag)
+    .reduce<Partial<Record<NumericModifier, number>>>(
+      (acc, tag) => ({
+        ...acc,
+        [MODIFIER_TAGS[tag]]:
+          ((acc[MODIFIER_TAGS[tag]] ?? 0) + 1) / tags.length,
```

Comment:
Technicality but is `tags.length` right here for `pokefam`-tag-having familiars?

Reply:
Could you just do the filter earlier to solve this?

`const tags = getFamiliarTags(familiar).filter(isStillsuitTag);`